File: fixtures/per-site-client/catalogClient.ts
// A client class given its base path when it is made, whose ten methods
// all send their request through one that builds the URL.

import { http } from "./http";

export class CatalogClient {
  constructor(private readonly baseUrl: string) {}

  private request(endpoint: string) {
    return http.get(`${this.baseUrl}/${endpoint}`);
  }

  listProducts() {
    return this.request("products");
  }

  listCategories() {
    return this.request("categories");
  }

  listBrands() {
    return this.request("brands");
  }

  listCollections() {
    return this.request("collections");
  }

  listReviews() {
    return this.request("reviews");
  }

  listVariants() {
    return this.request("variants");
  }

  listPrices() {
    return this.request("prices");
  }

  listStock() {
    return this.request("stock");
  }

  listTags() {
    return this.request("tags");
  }

  listBundles() {
    return this.request("bundles");
  }
}

File: fixtures/per-site-client-python/catalog_client.py
# A client class given its base path when it is made, whose ten methods
# all send their request through one that builds the URL.

import requests


class CatalogClient:
    def __init__(self, base_url):
        self.base_url = base_url

    def request(self, endpoint):
        return requests.get(f"{self.base_url}/{endpoint}")

    def list_products(self):
        return self.request("products")

    def list_categories(self):
        return self.request("categories")

    def list_brands(self):
        return self.request("brands")

    def list_collections(self):
        return self.request("collections")

    def list_reviews(self):
        return self.request("reviews")

    def list_variants(self):
        return self.request("variants")

    def list_prices(self):
        return self.request("prices")

    def list_stock(self):
        return self.request("stock")

    def list_tags(self):
        return self.request("tags")

    def list_bundles(self):
        return self.request("bundles")
